### ISSUE TITLE
[css-view-transitions-2] Rename `:active-view-transition()` pseudo-class

### ISSUE THREAD

**Issue by nt1m** [w3c/csswg-drafts#9972]
The current name is a bit confusing and makes it very tempting to put a view-transition name instead of type as argument. This is the type of thing that can typically confuse beginners very easily.

I'd like to suggest renaming to something like `:view-transition-type()` or such.

cc @vmpstr @noamr @khushalsagar

**Comment by khushalsagar** [comment]
I'd be ok with this name too.

FWIW, the rationale behind the "active" suffix in `active-view-transition` was that the pseudo-class only activates when a ViewTransition is running. For example, you could use `active-view-transition(*)` for styles which should apply only when a transition is running.

**Comment by noamr** [comment]
We've discussed this in the WG I belive, it's a bit weird when you want to use it without types, e.g. `:active-view-transition(*)`.

Perhaps we can do this?
```css
::root:active-view-transition { }
::root:view-transition-type(slide)
```

Happy to discuss at the WG if you want to raise this @nt1m

**Comment by nt1m** [comment]
Sure, I guess we can bikeshed on the call. I'd ideally want to not end up with 2 names, but my main concern is the confusion with the argument.

**Comment by bramus** [comment]
I’m OK with requiring an argument in the pseudo-class. Draws a parallel with the pseudo-elements

Don’t like the suggested `view-transition-type` as a name, as that would imply that you _must_ set a type before it can ever match (`*` is not the same as `null`).

What about a merger of both? I.e. allowing (requiring) an argument to `:active-view-transition`:
- `:active-view-transition(*)`
- `:active-view-transition(type)`

That would also need some specificity-rules like what we did with the pseudos.
- `*` = add 0,0,0
- Anything but `*` = add 0,0,1

**Comment by bramus** [comment]
> I think the problem with the name `active-view-transition` is the confusion on what actually goes in the argument.

Something like `:active-view-transition(type: <ident>)` would make this more clear, but I don’t think there’s a precedent in CSS that would allow named arguments in selectors. Also would not make sense to require the name if a-v-t only accepts 1 argument over time.

> A beginner mistake would be to put a view-transition-name instead of the transition `type` for instance.
> These kinds of mistakes on "similar but not quite exact things" are very easy to make by beginners.

That looks like something for DevRel / documentation to clear up.

**Comment by noamr** [comment]
> > I think the problem with the name `active-view-transition` is the confusion on what actually goes in the argument.
> 
> Something like `:active-view-transition(type: <ident>)` would make this more clear, but I don’t think there’s a precedent in CSS that would allow named arguments in selectors. 

Can be `:active-view-transition(type slide-in)`, similar to `:nth-child(2 of .class)`.

Also would not make sense to require the name if a-v-t only accepts 1 argument over time.
> 
> > A beginner mistake would be to put a view-transition-name instead of the transition `type` for instance.
> > These kinds of mistakes on "similar but not quite exact things" are very easy to make by beginners.
> 
> That looks like something for DevRel / documentation to clear up.

I tend to agree. It's a mistake that's easy to spot and fix.

**Comment by nt1m** [comment]
> That looks like something for DevRel / documentation to clear up.

I do not agree all the weight should be on DevRel / documentation, things should be easy to understand by design. DevRel / documentation should only be last resort to ensure good understanding of a feature.

**Comment by yisibl** [comment]
Do we need to consider using `<dashed-ident>` in function values to avoid conflicts with CSS keywords?

**Comment by khushalsagar** [comment]
^ that's a good point. The function takes in custom idents which can be used as types and we don't have any keywords for this property. So dashed-ident shouldn't be necessary here. @noamr for confirming.

**Comment by noamr** [comment]
> ^ that's a good point. The function takes in custom idents which can be used as types and we don't have any keywords for this property. So dashed-ident shouldn't be necessary here. @noamr for confirming.

Right, there is no added value in using dashed idents here at this point. People can use them if they like but it's not mandated
